modules/standard/Random.chpl

```diff
@@ -504,15 +583,92 @@ module Random {
 
 
     /*
-      Models a stream of pseudorandom numbers.  See the module-level
-      notes for :mod:`PCGRandom` for details on the PRNG used.
+
+      Models a stream of pseudorandom numbers generated by the PCG random number
+      generator.  See http://www.pcg-random.org/ and the paper, "PCG: A Family
+      of Simple Fast Space-Efficient Statistically Good Algorithms for Random
+      Number Generation" by M.E. O'Neill.
+
+      This class builds upon the :record:`pcg_setseq_64_xsh_rr_32_rng` PCG RNG
+      which has 64 bits of state and 32 bits of output.
+
+      While the PCG RNG used here is believed to have good statistical
+      properties, it is not suitable for cryptographic use. Additionally,
+      if statistical properties of the random numbers are very important,
+      another strategy may be required.
+
+      We have verified that the random numbers generated by this class can match
+      those generated by the C PCG reference implementation. However, this
+      implementation differs from the C PCG reference implementation in how it
+      produces random integers in a range and in support for generating random
+      `real` s.
+
+      Smaller numbers, such as uint(8) or uint(16), are generated from
+      the high-order bits of the 32-bit output.
+
+      To generate larger numbers, several 32-bit-output RNGs are ganged
+      together.  This strategy is recommended by the author of PCG (and
+      demonstrated in the file `pcg32x2-demo.c`. Each of these 32-bit RNGs has a
+      different sequence constant and so will be independent and uncorrelated.
+      For example, to generate 128-bit complex numbers, this RNG will use
+      4 ganged 32-bit PCG RNGs with different sequence constants. One impact of
+      this approach is that this implementation will only generate 2**64
+      different complex numbers (for example).
+
+      This class also supports generating integers within particular bounds.
+      When that is required, this class uses a strategy different from the PCG
+      reference implementation in order to work better a parallel setting. In
```

Comment:
typo: "in a parallel setting"

I continue to be unconvinced that we should be in the business of providing bounded random integers in a parallel setting within RandomStream itself...